aws-api/src/test/java/com/amplifyframework/api/aws/auth/OwnerBasedAuthTest.java

```diff
@@ -0,0 +1,351 @@
+                        .put("authorizationType", "AMAZON_COGNITO_USER_POOLS"))
+                .put(GRAPHQL_API_WITH_OIDC, new JSONObject()
+                        .put("endpointType", "GraphQL")
+                        .put("endpoint", baseUrl.url())
+                        .put("region", "us-east-1")
+                        .put("authorizationType", "OPENID_CONNECT"));
+        } catch (JSONException exception) {
+            // This shouldn't happen...
+        }
+
+        plugin = new AWSApiPlugin(providers);
+        plugin.configure(configuration, ApplicationProvider.getApplicationContext());
+    }
+
+    /**
+     * Test that owner argument fails to be appended to subscription request if
+     * the authorization mode is not OIDC compliant.
+     */
+    @Test
+    public void ownerArgumentNotAddedWithApiKey() {
+        // Set API to use API key auth mode
+        apiName = GRAPHQL_API_WITH_API_KEY;
+
+        // Attempting to subscribe to a model with owner-based auth with API key auth mode.
+        GraphQLRequest<OwnerAuth> request = ModelSubscription.onCreate(OwnerAuth.class);
+        GraphQLOperation<OwnerAuth> operation = subscribe(request);
+
+        // Subscription should fail at pre-processing
+        assertNull(operation);
+    }
+
+    /**
+     * Test that request is serialized as expected, with owner variable.
+     * @throws JSONException from JSONAssert.assertEquals
+     */
+    @Test
+    public void ownerArgumentIsAddedAndSerializedInRequest() throws JSONException {
+        // Set API to use Cognito User Pools auth mode
+        apiName = GRAPHQL_API_WITH_COGNITO;
+
+        GraphQLRequest<OwnerAuth> request = ModelSubscription.onCreate(OwnerAuth.class);
+        GraphQLOperation<OwnerAuth> operation = subscribe(request);
+
+        assertNotNull(operation);
+        JSONAssert.assertEquals(Resources.readAsString("request-owner-auth.json"),
+                operation.getRequest().getContent(),
+                true);
+    }
+
+    /**
+     * Verify that owner argument is required for all subscriptions if ModelOperation.READ is specified
+     * while using Cognito User Pools auth mode.
+     * @throws AmplifyException if a ModelSchema can't be derived from the Model class.
+     */
+    @Test
+    public void ownerArgumentAddedForRestrictedReadWithUserPools() throws AmplifyException {
+        // Set API to use Cognito User Pools auth mode
+        apiName = GRAPHQL_API_WITH_COGNITO;
+
+        assertTrue(isOwnerArgumentAdded(Owner.class, SubscriptionType.ON_UPDATE));
+        assertTrue(isOwnerArgumentAdded(OwnerRead.class, SubscriptionType.ON_UPDATE));
+
+        assertTrue(isOwnerArgumentAdded(Owner.class, SubscriptionType.ON_DELETE));
+        assertTrue(isOwnerArgumentAdded(OwnerRead.class, SubscriptionType.ON_DELETE));
+
+        assertTrue(isOwnerArgumentAdded(Owner.class, SubscriptionType.ON_CREATE));
+        assertTrue(isOwnerArgumentAdded(OwnerRead.class, SubscriptionType.ON_CREATE));
```

Comment:
Looks like it could be a `for` (or `for`-`for`) loop?

Reply:
This was not originally written by me, but I think this format improves readability.